src/backend/cdb/cdbmetadatacache.c

```diff
@@ -619,6 +619,9 @@ BlockLocation *GetHdfsFileBlockLocations(const HdfsFileInfo *file_info, uint64_t
             if (cache_entry->block_num <= 1)
             {
                 // only one file, re-fetch 
+                LWLockRelease(MetadataCacheLock); 
+                
+                LWLockAcquire(MetadataCacheLock, LW_EXCLUSIVE);
```

Comment:
Hi George,

Thanks for your comment. It only has one lock (MetadataCacheLock) and one resource (MetadataCache) which lock on, so it should not cause dead lock.